lib/DriverTool/autolink_extract_main.cpp

```diff
@@ -140,8 +142,15 @@ extractLinkerFlagsFromObjectFile(const llvm::object::ObjectFile *ObjectFile,
       llvm::SmallVector<llvm::StringRef, 4> SplitFlags;
       SectionData->split(SplitFlags, llvm::StringRef("\0", 1), -1,
                          /*KeepEmpty=*/false);
-      for (const auto &Flag : SplitFlags)
-        LinkerFlags.push_back(Flag.str());
+      for (const auto &Flag : SplitFlags) {
+        // If this is a library '-lxxx' flag, only add it if we have not seen it before
+        if (Flag.str().rfind("-l", 0) == 0) {
```

Comment:
Why `std::string::rfind` over `std::string::compare` which would be more idiomatic and clearer?